### PR TITLE
MSDK-6849 -  Update the MSDK to 10.0.0

### PR DESCRIPTION
**Pull request by nkosta**
- Updates the MSDK to 10.0.0 and adapts the code to Swift Concurrency.

### CODE REVIEW

File: IntegrationExample/AppDelegate.swift [hunk 1]
```diff
@@ -6,6 +6,7 @@
 //
 
 import Foundation
+import SwiftUI
```
Comment (SalarZ) [inline IntegrationExample/AppDelegate.swift:9]: We don't need to import SwiftUI here.

File: IntegrationExample/AppDelegate.swift [hunk 2]
```diff
@@ -71,13 +74,19 @@ final class AppDelegate: NSObject, UIApplicationDelegate {
...
+                                Personalization {
+                                    FineTuning(fitting: fitting)
+                                }
+                            }
+
+        return try await MimiCore.shared.processing.activate(configuration: configuration)
+
         // If only 1 preset, is desired, the Default datasource is to be used as follows.
 //        let fitting = MimiPersonalization.Fitting.techLevel(techLevel)
 //        return try processing.activate(presetDataSource: .default(.init(fitting: fitting)))
```
Comment (SalarZ) [inline IntegrationExample/AppDelegate.swift:92]: We need to change it to use the configuration with `SinglePreset`.

File: IntegrationExample/Processing/PartnerAudioProcessingController.swift [hunk 3]
```diff
@@ -6,137 +6,91 @@
 //
 
 import Foundation
+import SwiftUI
```
Comment (SalarZ) [inline IntegrationExample/Processing/PartnerAudioProcessingController.swift:9]: We don't need to import SwiftUI here.

File: IntegrationExample/Processing/PartnerAudioProcessingController.swift [hunk 4]
```diff
@@ -6,137 +6,91 @@
...
+
+            // If for any reason e.g. Mimi enabled headphones disconnected, the `value` cannot
+            // be applied then throw an error here.
+
+            try await self.firmwareController.setIsEnabled(value)
+        })
+
+        do {
+            // Sync the applicator with the firmware isEnabled value.
+            try await session.isEnabled.apply(isEnabledValue)
```
Comment (SalarZ) [inline IntegrationExample/Processing/PartnerAudioProcessingController.swift:51]: I have a question: where and when do we need to use the `synchronizeApplicators` function?

File: IntegrationExample/Processing/View/ProcessingViewModel.swift [hunk 5]
```diff
@@ -31,67 +32,71 @@ final class ProcessingViewModel: ObservableObject {
...
+        core.processing.sessionPublisher
+            .receive(on: DispatchQueue.main)
             .sink { [weak self] session in
                 guard let session else {
                     self?.isSessionAvailable = false
                     return
                 }
-                // Since the Slider sends updates continously, setting the delivery mode to discreet
-                // allows us to debounce the updates.
-                session.intensity.deliveryMode = .discrete(seconds: 0.2)
```
Comment (SalarZ) [inline IntegrationExample/Processing/View/ProcessingViewModel.swift:66]: As we don't have the `deliveryMode` anymore, I think it's good to set the `uiControlDebounceBehavior` in the configuration.

**Review by SalarZ — CHANGES_REQUESTED** [review]
Just a few comments

**Comment by hozefa26** [comment]
@SalarZ since kosta is out this week, can you please make the changes you suggested and then i'll take a look at it?

**Review by hozefa26 — APPROVED** [review]
just pushed some minor changes but otherwise great work @SalarZ & @nkosta! 🚀

**Review by SalarZ — APPROVED** [review]
🚀 Thanks @hozefa26 for cleaning up.

**Pull request merged**